packages/core/src/parseAmplifyGen2.ts

```diff
@@ -0,0 +1,245 @@
+		}
+
+		config.Auth.Cognito.loginWith = {};
+
+		if (hasOAuthConfig(authProperties)) {
+			config.Auth.Cognito.loginWith!.oauth = {
+				domain: authProperties.oauth_domain!,
+				redirectSignIn: authProperties.oauth_redirect_sign_in!,
+				redirectSignOut: authProperties.oauth_redirect_sign_out!,
+				responseType: authProperties.oauth_response_type!,
+				scopes: authProperties.oauth_scopes!,
+				providers: getOAuthProviders(authProperties.identity_providers),
+			};
+		}
+
+		if (authProperties.user_verification_mechanisms) {
+			if (
+				authProperties.user_verification_mechanisms.some(
+					user => user === 'EMAIL',
+				)
+			) {
+				config.Auth.Cognito.loginWith.email = true;
+			}
+
+			if (
+				authProperties.user_verification_mechanisms.some(
+					user => user === 'PHONE',
+				)
+			) {
+				config.Auth.Cognito.loginWith.phone = true;
+			}
+
+			if (
+				authProperties.user_verification_mechanisms.some(
+					user => user === 'USERNAME',
+				)
+			) {
+				config.Auth.Cognito.loginWith.username = true;
+			}
+		}
+
+		if (authProperties.standard_attributes) {
+			config.Auth.Cognito.userAttributes = authProperties.standard_attributes;
+		}
+
+		if (config.Auth.Cognito.loginWith) {
+			delete config.Auth.Cognito.loginWith;
+		}
+	}
+
+	if (resourceConfig.analytics) {
+		const analyticsProperties = resourceConfig.analytics;
+
+		if (analyticsProperties.amazon_pinpoint) {
+			const Pinpoint = {
+				appId: analyticsProperties.amazon_pinpoint.app_id,
+				region: analyticsProperties.amazon_pinpoint.aws_region,
+			};
+			if (!config.Analytics) {
+				config.Analytics = {
+					Pinpoint,
+				};
+			} else {
+				config.Analytics!.Pinpoint = Pinpoint;
+			}
+		}
+
+		if (analyticsProperties.amazon_kinesis) {
+			const Kinesis = {
+				region: analyticsProperties.amazon_kinesis.aws_region,
+				bufferSize: analyticsProperties.amazon_kinesis.buffer_size,
+				flushInterval: analyticsProperties.amazon_kinesis.flush_interval,
+				flushSize: analyticsProperties.amazon_kinesis.flush_size,
+				resendLimit: analyticsProperties.amazon_kinesis.resend_limit,
+			};
+
+			if (!config.Analytics) {
+				config.Analytics = {
+					Kinesis,
+				};
+			} else {
+				config.Analytics!.Kinesis = Kinesis;
+			}
+		}
+
+		if (analyticsProperties.amazon_kinesis_firehose) {
+			const KinesisFirehose = {
+				region: analyticsProperties.amazon_kinesis_firehose.aws_region,
+				bufferSize: analyticsProperties.amazon_kinesis_firehose.buffer_size,
+				flushInterval:
+					analyticsProperties.amazon_kinesis_firehose.flush_interval,
+				flushSize: analyticsProperties.amazon_kinesis_firehose.flush_size,
+				resendLimit: analyticsProperties.amazon_kinesis_firehose.resend_limit,
+			};
+
+			if (!config.Analytics) {
+				config.Analytics = {
+					KinesisFirehose,
+				};
+			} else {
+				config.Analytics!.KinesisFirehose = KinesisFirehose;
+			}
+		}
+	}
+
+	if (resourceConfig.geo) {
+		const geoProperties = resourceConfig.geo;
```

Comment:
Nit: Looks like we can save some duplication and improve readability a bit if we destructure. I think this is broadly applicable to all of the other category configs above, especially the analytics ones where we have to drill multiple levels into provider configs.
```suggestion
		const { aws_region, geofence_collections, maps, search_indices } = resourceConfig.geo;
```